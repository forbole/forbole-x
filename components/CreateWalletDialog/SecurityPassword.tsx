import { Button, DialogActions, DialogContent, DialogContentText, Box } from '@material-ui/core';
import useTranslation from 'next-translate/useTranslation';
import React from 'react';
import PasswordInput from '../PasswordInput';
import useStyles from './styles';

interface SecurityPasswordProps {
  onConfirm(password: string): void;
}

const SecurityPassword: React.FC<SecurityPasswordProps> = ({ onConfirm }) => {
  const { t } = useTranslation('common');
  const classes = useStyles();
  const [password, setPassword] = React.useState('');

  return (
    <form
      noValidate
      onSubmit={e => {
        e.preventDefault();
        onConfirm(password);
      }}>
      <DialogContent className={classes.dialogContent}>
        <DialogContentText>{t('security password description')}</DialogContentText>
        <Box mb={20}>
          <PasswordInput
            placeholder={t('password')}
            value={password}
            onChange={e => setPassword(e.target.value)}
            withSecurityLevel
          />
        </Box>
      </DialogContent>
      <DialogActions>
        <Box flex={1} display="flex" flexDirection="column" mb={3}>
          <Button
            className={classes.button}
            variant="contained"
            color="primary"
            disabled={password.length < 6}
            type="submit">
            {t('next')}
          </Button>
        </Box>
      </DialogActions>
    </form>
  );
};

export default SecurityPassword;
