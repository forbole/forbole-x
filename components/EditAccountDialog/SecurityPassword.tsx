import {
  Button,
  DialogActions,
  DialogContent,
  DialogContentText,
  Box,
  CircularProgress,
  useTheme,
} from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'
import PasswordInput from '../PasswordInput'
import useStyles from './styles'

interface SecurityPasswordProps {
  onConfirm(password: string): void
  loading: boolean
}

const SecurityPassword: React.FC<SecurityPasswordProps> = ({ onConfirm, loading }) => {
  const { t } = useTranslation('common')
  const classes = useStyles()
  const theme = useTheme()
  const [password, setPassword] = React.useState('')

  return (
    <form
      noValidate
      onSubmit={(e) => {
        e.preventDefault()
        onConfirm(password)
      }}
    >
      <DialogContent className={classes.dialogContent}>
        <DialogContentText>{t('enter security password')}</DialogContentText>
        <Box mb={30}>
          <PasswordInput
            placeholder={t('password')}
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
        </Box>
      </DialogContent>
      <DialogActions>
        <Box flex={1} display="flex" flexDirection="column" mb={3}>
          <Button
            className={classes.fullWidthButton}
            variant="contained"
            color="primary"
            disabled={!password.length || loading}
            type="submit"
          >
            {loading ? <CircularProgress size={theme.spacing(3.5)} /> : t('confirm')}
          </Button>
        </Box>
      </DialogActions>
    </form>
  )
}

export default SecurityPassword
