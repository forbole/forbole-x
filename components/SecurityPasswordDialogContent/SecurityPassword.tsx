import {
  Button,
  DialogActions,
  DialogContent,
  DialogContentText,
  CircularProgress,
  useTheme,
} from '@material-ui/core';
import useTranslation from 'next-translate/useTranslation';
import React from 'react';
import useStyles from './styles';
import PasswordInput from '../PasswordInput';
import { useWalletsContext } from '../../contexts/WalletsContext';

interface SecurityPasswordProps {
  onForgot: () => void;
  onUnlock: (password: string) => void;
  password: string;
  setPassword: any;
  loading: boolean;
  walletId: string;
}

const SecurityPassword: React.FC<SecurityPasswordProps> = ({
  onForgot,
  onUnlock,
  password,
  setPassword,
  loading,
  walletId,
}) => {
  const { t } = useTranslation('common');
  const classes = useStyles();
  const [error, setError] = React.useState('');
  const theme = useTheme();
  const { viewMnemonicPhrase } = useWalletsContext();

  const confirm = React.useCallback(async () => {
    try {
      await viewMnemonicPhrase(walletId, password);
      onUnlock(password);
    } catch (err) {
      setError(t(err.message));
    }
  }, [walletId, onUnlock, password]);

  return (
    <form
      noValidate
      className={classes.form}
      onSubmit={e => {
        e.preventDefault();
        confirm();
      }}
    >
      <DialogContent>
        <DialogContentText>{t('enter security password')}</DialogContentText>
        <PasswordInput
          autoFocus
          placeholder={t('password')}
          value={password}
          error={!!error}
          helperText={error}
          onChange={e => setPassword(e.target.value)}
        />
      </DialogContent>
      <DialogActions className={classes.action}>
        <Button
          className={classes.button}
          variant="contained"
          color="primary"
          disabled={!password.length || loading}
          type="submit"
        >
          {loading ? <CircularProgress size={theme.spacing(3.5)} /> : t('next')}
        </Button>
        <Button className={classes.forgotButton} onClick={onForgot}>
          {t('forgot password?')}
        </Button>
      </DialogActions>
    </form>
  );
};

export default SecurityPassword;
