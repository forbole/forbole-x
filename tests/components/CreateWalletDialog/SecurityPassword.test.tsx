import React from 'react';
import renderer from 'react-test-renderer';
import SecurityPassword from '../../../components/CreateWalletDialog/SecurityPassword';

const onConfirm = jest.fn();

jest.mock('../../../components/PasswordInput', () => props => (
  <div id="PasswordInput" {...props} />
));

describe('component: CreateWalletDialog - SecurityPassword', () => {
  it('renders correctly', () => {
    const component = renderer.create(<SecurityPassword onConfirm={onConfirm} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with password less than 6 char', () => {
    const component = renderer.create(<SecurityPassword onConfirm={onConfirm} />);
    renderer.act(() => {
      component.root
        .findByProps({
          id: 'PasswordInput',
        })
        .props.onChange({
          target: {
            value: '123',
          },
        });
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with weak password', () => {
    const component = renderer.create(<SecurityPassword onConfirm={onConfirm} />);
    renderer.act(() => {
      component.root
        .findByProps({
          id: 'PasswordInput',
        })
        .props.onChange({
          target: {
            value: '123456',
          },
        });
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with medium strength password', () => {
    const component = renderer.create(<SecurityPassword onConfirm={onConfirm} />);
    renderer.act(() => {
      component.root
        .findByProps({
          id: 'PasswordInput',
        })
        .props.onChange({
          target: {
            value: '123123qwe',
          },
        });
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with strong password', () => {
    const component = renderer.create(<SecurityPassword onConfirm={onConfirm} />);
    renderer.act(() => {
      component.root
        .findByProps({
          id: 'PasswordInput',
        })
        .props.onChange({
          target: {
            value: '123123qweQWE!@#',
          },
        });
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('calls onConfirm when button is clicked', () => {
    const component = renderer.create(<SecurityPassword onConfirm={onConfirm} />);
    renderer.act(() => {
      component.root
        .findByProps({
          id: 'PasswordInput',
        })
        .props.onChange({
          target: {
            value: '123123qweQWE!@#',
          },
        });
    });
    renderer.act(() => {
      component.root.findByType('form').props.onSubmit({ preventDefault: jest.fn() });
    });
    expect(onConfirm).toBeCalledWith('123123qweQWE!@#');
  });
});

afterEach(() => {
  jest.clearAllMocks();
});
