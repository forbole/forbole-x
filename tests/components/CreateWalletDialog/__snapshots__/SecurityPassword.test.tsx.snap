// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: CreateWalletDialog - SecurityPassword renders correctly 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-7"
  >
    <p
      className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      common:security password description
    </p>
    <div
      className="MuiBox-root MuiBox-root-9"
    >
      <div
        id="PasswordInput"
        onChange={[Function]}
        placeholder="common:password"
        value=""
        withSecurityLevel={true}
      />
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <div
      className="MuiBox-root MuiBox-root-10"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-1 MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          common:next
        </span>
      </button>
    </div>
  </div>
</form>
`;

exports[`component: CreateWalletDialog - SecurityPassword renders correctly with medium strength password 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-7"
  >
    <p
      className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      common:security password description
    </p>
    <div
      className="MuiBox-root MuiBox-root-15"
    >
      <div
        id="PasswordInput"
        onChange={[Function]}
        placeholder="common:password"
        value="123123qwe"
        withSecurityLevel={true}
      />
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <div
      className="MuiBox-root MuiBox-root-16"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-1 MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          common:next
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</form>
`;

exports[`component: CreateWalletDialog - SecurityPassword renders correctly with password less than 6 char 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-7"
  >
    <p
      className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      common:security password description
    </p>
    <div
      className="MuiBox-root MuiBox-root-11"
    >
      <div
        id="PasswordInput"
        onChange={[Function]}
        placeholder="common:password"
        value="123"
        withSecurityLevel={true}
      />
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <div
      className="MuiBox-root MuiBox-root-12"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-1 MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          common:next
        </span>
      </button>
    </div>
  </div>
</form>
`;

exports[`component: CreateWalletDialog - SecurityPassword renders correctly with strong password 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-7"
  >
    <p
      className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      common:security password description
    </p>
    <div
      className="MuiBox-root MuiBox-root-17"
    >
      <div
        id="PasswordInput"
        onChange={[Function]}
        placeholder="common:password"
        value="123123qweQWE!@#"
        withSecurityLevel={true}
      />
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <div
      className="MuiBox-root MuiBox-root-18"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-1 MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          common:next
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</form>
`;

exports[`component: CreateWalletDialog - SecurityPassword renders correctly with weak password 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-7"
  >
    <p
      className="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      common:security password description
    </p>
    <div
      className="MuiBox-root MuiBox-root-13"
    >
      <div
        id="PasswordInput"
        onChange={[Function]}
        placeholder="common:password"
        value="123456"
        withSecurityLevel={true}
      />
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <div
      className="MuiBox-root MuiBox-root-14"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-1 MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          common:next
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</form>
`;
